Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -105,27 +123,132 @@ public override bool ExportExecutionDetailsToALM(BusinessFlow bizFlow, ref strin
+                fields.Append(
+                    AddFieldsValues(octaneRepository.GetEntityFields(resourse, loginDto), "Test Suit",
+                    listnodes, octaneRepository.GetPhases(loginDto, resourse)));
+
+                resourse = Octane_Repository.BLL.Extensions.ConvertResourceType(ResourceType.REQUIREMENT);
+                fields.Append(
+                    AddFieldsValues(
+                        octaneRepository.GetEntityFields(resourse, loginDto), "Requirement",
+                        listnodes, octaneRepository.GetPhases(loginDto, resourse)));
+
+                resourse = Octane_Repository.BLL.Extensions.ConvertResourceType(ResourceType.TEST_RUN);
+                fields.Append(
+                    AddFieldsValues(
+                        octaneRepository.GetEntityFields(resourse, loginDto), "Run",
+                        listnodes, octaneRepository.GetPhases(loginDto, resourse)));
+            }
+            else
+            {
+
+                resourse = Octane_Repository.BLL.Extensions.ConvertResourceType(resourceType);
+
+                fields.Append(AddFieldsValues(Task.Run(() =>
+                {
+                    return octaneRepository.GetEntityFields(resourse, loginDto);
+                }).Result, resourse, listnodes, Task.Run(() =>
+                {
+                    return octaneRepository.GetPhases(loginDto, resourse);
+                }).Result));
+            }
+            return fields;
+        }
+
+        private static ObservableList<ExternalItemFieldBase> AddFieldsValues(ListResult<FieldMetadata> entityFields, string entityType, Dictionary<string, List<string>> listnodes, Dictionary<string, List<string>> phases)
+        {
+            ObservableList<ExternalItemFieldBase> fields = new ObservableList<ExternalItemFieldBase>();
+
+            if ((entityFields != null) && (entityFields.total_count.Value > 0))
+            {
+                foreach (FieldMetadata field in entityFields.data)
+                {
+                    if (string.IsNullOrEmpty(field.Label) || !field.VisibleInUI)
+                    {
+                        continue;
+                    }
+
+                    ExternalItemFieldBase itemfield = new ExternalItemFieldBase();
+                    itemfield.ID = field.Label;
+                    itemfield.ExternalID = field.Label;
+                    itemfield.Name = field.Name;
+                    itemfield.Mandatory = field.IsRequired;
+                    itemfield.SystemFieled = !field.IsUserField;
+                    
+                    if (itemfield.Mandatory)
+                    {
+                        itemfield.ToUpdate = true;
+                    }
+                    itemfield.ItemType = entityType;
+                    itemfield.Type = field.FieldType;
+
+                    if (listnodes != null && listnodes.ContainsKey(field.Name) && listnodes[field.Name].Any())
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6235433)